include/zephyr/modem/chat.h

```diff
@@ -376,6 +376,156 @@ void modem_chat_script_abort(struct modem_chat *chat);
  */
 void modem_chat_release(struct modem_chat *chat);
 
+/**
+ * @brief Initialize modem chat match
+ * @param chat_match Modem chat match instance
+ */
+void modem_chat_match_init(struct modem_chat_match *chat_match);
+
+/**
+ * @brief Set match of modem chat match instance
+ * @param chat_match Modem chat match instance
+ * @param match Match to set
+ * @note The lifetime of match must match or exceed the lifetime of chat_match
+ * @warning Always call this API after match is modified
+ *
+ * @retval 0 if successful
+ * @retval -ENOMEM if strlen of match exceeds UINT8_MAX
+ */
+int modem_chat_match_set_match(struct modem_chat_match *chat_match, const char *match);
+
+/**
+ * @brief Set separators of modem chat match instance
+ * @param chat_match Modem chat match instance
+ * @param separators Separators to set
+ * @note The lifetime of separators must match or exceed the lifetime of chat_match
+ * @warning Always call this API after separators is modified
+ *
+ * @retval 0 if successful
+ * @retval -ENOMEM if strlen of separators exceeds UINT8_MAX
+ */
+int modem_chat_match_set_separators(struct modem_chat_match *chat_match, const char *separators);
+
+/**
+ * @brief Set modem chat match callback
+ * @param chat_match Modem chat match instance
+ * @param callback Callback to set
+ */
+void modem_chat_match_set_callback(struct modem_chat_match *chat_match,
+				   modem_chat_match_callback callback);
+
+/**
+ * @brief Set modem chat match partial flag
+ * @param chat_match Modem chat match instance
+ * @param partial Partial flag to set
+ */
+void modem_chat_match_set_partial(struct modem_chat_match *chat_match, bool partial);
+
+/**
+ * @brief Set modem chat match wildcards flag
+ * @param chat_match Modem chat match instance
+ * @param wildcards Wildcards flag to set
+ */
+void modem_chat_match_set_wildcards(struct modem_chat_match *chat_match, bool wildcards);
```

Comment:
Not introduced by this PR, but I think that `wildcards` would benefit from an improved naming, e.g. `uses_wildcards`, `wildcard_enabled`... Because currently the naming could mean that what you are dealing with is the actual wildcards (although it becomes obvious that it's not the case when you have a closer look).
Looking at it, why not also declaring it (and `partial`) as bool instead of `uint8_t : 1`? :slightly_smiling_face:

Reply:
Agreed, changed the signature to:
```
void modem_chat_match_enable_wildcards(struct modem_chat_match *chat_match, bool enable);
```
this should read correctly :) Also changed the members  to `bool`